roles/grafana/templates/grafana.ini.j2

```diff
@@ -3,18 +3,22 @@
 # http://docs.grafana.org/installation/configuration
 # https://github.com/grafana/grafana/blob/master/conf/sample.ini
 
-{% for k, v in grafana_ini.items() %}
-{% if v is not mapping %}
-{{ k }} = {{ v }}
-{% endif %}
-{% endfor %}
-
 {% for section, items in grafana_ini.items() %}
 {% if items is mapping %}
 [{{ section }}]
-{% for k,v in items.items() %}
+{% for sub_key, sub_value in items.items() %}
+{% if sub_value is mapping %}
+
+[{{ section }}.{{ sub_key }}]
+{% for k, v in sub_value.items() %}
 {{ k }} = {{ v }}
 {% endfor %}
-
+{% else %}
+{{ sub_key }} = {{ sub_value }}
+{% endif %}
+{% endfor %}
+{% else %}
+{{ section }} = {{ items }}
```

Comment:
Or can sectionless keys be set here  `if item is not mapping`?

They need to be at the top before any sections to be sectionless. I don't suppose the `not mapping` items are guaranteed to be iterated first?

Reply:
https://github.com/grafana/grafana-ansible-collection/pull/303